test/e2e/helpers.go

```diff
@@ -2,65 +2,166 @@ package e2e
+	pc := NewPrometheusInfo(caCtx)
+	path := fmt.Sprintf("%s%s", pc.queryPath, url.QueryEscape(fmt.Sprintf(`knative_up{type="%s"}`, label)))
+	metricsURL, err := url.Parse(path)
 	if err != nil {
 		caCtx.T.Fatalf("Error parsing url for metrics: %v", err)
 	}
-	expectedStr := fmt.Sprintf(`knative_up{type="%s"} %d`, metricLabel, expectedValue)
 	// Wait until the endpoint is actually working and we get the expected value back
 	_, err = pkgTest.WaitForEndpointState(
 		context.Background(),
 		caCtx.Clients.Kube,
 		caCtx.T.Logf,
 		metricsURL,
-		pkgTest.EventuallyMatchesBody(expectedStr),
+		eventuallyMatchesValue(expectedValue),
 		"WaitForMetricsToServeText",
-		true)
+		true, pc.options...)
+	if err != nil {
+		caCtx.T.Fatalf("Failed to access the Prometheus API endpoint and get the metric value expected: %v", err)
+	}
+}
+
+func VerifyServingControlPlaneMetrics(caCtx *test.Context) {
+	pc := NewPrometheusInfo(caCtx)
+	servingMetrics := []string{
+		"activator_client_results",
+		"autoscaler_actual_pods",
+		"hpaautoscaler_client_latency_bucket",
+		"controller_client_latency_bucket",
+		"domainmapping_client_latency_bucket",
+		"domainmapping_webhook_client_latency_bucket",
+		"webhook_client_latency_bucket",
+	}
+	for _, metric := range servingMetrics {
+		path := fmt.Sprintf("%s%s", pc.queryPath, metric)
+		metricsURL, err := url.Parse(path)
+		if err != nil {
+			caCtx.T.Fatalf("Error parsing url for metrics %v", err)
+		}
+		// Wait until the endpoint is actually working and we get the expected value back
+		_, err = pkgTest.WaitForEndpointState(
+			context.Background(),
+			caCtx.Clients.Kube,
+			caCtx.T.Logf,
+			metricsURL,
+			pkgTest.EventuallyMatchesBody(metric),
+			"WaitForMetricsToServeText",
+			true, pc.options...)
+		if err != nil {
+			caCtx.T.Fatalf("Failed to access the Prometheus API endpoint and get the metric value expected: %v", err)
+		}
+	}
+}
+
+func getBearerTokenForPrometheusAccount(caCtx *test.Context) string {
+	sa, err := caCtx.Clients.Kube.CoreV1().ServiceAccounts("openshift-monitoring").Get(context.Background(), "prometheus-k8s", meta.GetOptions{})
+	if err != nil {
+		caCtx.T.Fatalf("Error getting service account prometheus-k8s: %v", err)
+	}
+	tokenSecret := getSecretNameForToken(sa.Secrets)
+	if tokenSecret == nil {
+		caCtx.T.Fatal("Token name for prometheus-k8s service account not found")
+	}
+	sec, err := caCtx.Clients.Kube.CoreV1().Secrets("openshift-monitoring").Get(context.Background(), *tokenSecret, meta.GetOptions{})
+	if err != nil {
+		caCtx.T.Fatalf("Error getting secret %s: %v", *tokenSecret, err)
+	}
+	tokenContents := sec.Data["token"]
+	if len(tokenContents) == 0 {
+		caCtx.T.Fatalf("Token data is missing for token %s", *tokenSecret)
+	}
+	return string(tokenContents)
+}
+
+func getSecretNameForToken(secrets []corev1.ObjectReference) *string {
+	for _, sec := range secrets {
+		if strings.Contains(sec.Name, "token") {
+			return &sec.Name
```

Comment:
nit: I'd return an empty string on "not found" and not make this a pointer.

Reply:
It is the old question if we should return a pointer for string, faced it upstream, the truth is k8s code base uses it a lot, and it is not a anti-pattern, I dont think it needs a change.

Reply:
wfm!